shared/AppInsightsCore/src/Config/DynamicProperty.ts

```diff
@@ -124,23 +144,35 @@ function _makeDynamicProperty<T, C, V = any>(state: _IDynamicConfigHandlerState<
     }
 
     objDefineAccessors(theConfig, detail.n, _getProperty, _setProperty, true);
+
+    // Return the dynamic reference
+    return _getProperty();
 }
 
-export function _setDynamicProperty<T, C, V = any>(state: _IDynamicConfigHandlerState<T>, target: C, name: string, value: V) {
+export function _setDynamicProperty<T, C, V = any>(state: _IDynamicConfigHandlerState<T>, target: C, name: string, value: V, inPlace?: boolean, rdOnly?: boolean): V {
     if (target) {
-        let isDynamic = false;
         // To be a dynamic property it needs to have a get function
-        let propDesc = objGetOwnPropertyDescriptor(target, name);
-        if (propDesc && propDesc.get) {
-            isDynamic = !!propDesc.get[state.prop];
-        }
+        let getter = _getOwnPropGetter(target, name);
+        let isDynamic = getter && !!getter[state.prop];
     
         if (!isDynamic) {
-            _makeDynamicProperty(state, target, name, value);
+            value = _makeDynamicProperty(state, target, name, value);
+            if (inPlace || rdOnly) {
+                getter = _getOwnPropGetter(target, name);
+            }
         } else {
-            // Looks like it's already dynamic and a different value so just assign the new value
+            // Looks like it's already dynamic just assign the new value
             target[name] = value;
         }
+
+        // Assign the optional flags if true
```

Comment:
By-design that once "marked" as referenced or read-only there is no going back (ie. there is no `unref` or `read-write` option).